scripts/pull-asset-registry.js

```diff
@@ -16,14 +16,10 @@ async function main() {
     writeToFile(lightNitePath, lightNiteAssetHashes);
 
     // Blockstream
-    const blockstreamAssetHashes = Object.values(res.data)
-      .filter(asset => asset.entity.domain.includes("blockstream.com"))
-      .map(asset => asset.asset_id);
-
+    res = await axios.get('https://raw.githubusercontent.com/Blockstream/asset_registry_db/master/icons.json');
+    const blockstreamAssetHashes = Object.keys(res.data);
```

Comment:
mmm blockstreamAssetHashes means they are issued by blockstream hence before I was checking in the domain field of the asset if is from them

Reply:
Yes, with the only purpose of having a default icon ('blockstream.png') for each asset where we didn't have the original icon. But, previously this would find 17 assets which now are ALL present in the new [json endpoint](https://raw.githubusercontent.com/Blockstream/asset_registry_db/master/icons.json), so we don't need this default icon logic anymore.

So now this variable means "icons we have from mempool because they support all blockstream asset hashes present in the new json endpoint".

Reply:
Yes so the variable name should change imho, but maybe what we should do is dropping this script entirely? I mean if we can now dinamically fetch an icon via network call, no need to bundle anything imho

Reply:
We need to know which asset hashes are available on https://liquid.network/api/v1/asset in order to implement the logic of showing an UNKNOW_IMG in the case they are not (see [getAssetImage()](https://github.com/bordalix/marina/blob/use_icons_from_mempool_334/src/application/utils/constants.ts#L65))

Reply:
What if we show Unknown Image anytime we either a) Get a 404 not found error b) the API is down?

I guess I would only retain hardcoded fallback for the real featured one (L-BTC & USDT) that's it, to lower the bundle as much as possible?

Reply:
Hummmm, that would mean getAssetImage() would have to make a fetch for the image, wait for the return value, and then return the image path based on the return code from the fetch. But then the browser would make the fetch also, which would mean double hits for every image. Not very elegant.

What we can do, though, is to use something of this kind on every image tag:

```
<img src="https://remote.path/image.png" onerror="this.src='images/unknown.png'">
```

What do you think?

Reply:
I like it! let's do it (it removes code and complexity and rely on native browser apis)

I would still pass a function to `src` where it will check if the asset hash is the featured ones (LBTC, USDT) and save a network call?

Reply:
Fixed by dacac83

Assets size reduction:
  index.js (2.68 MiB) => 2.66 MiB
  background-script.js (1.34 MiB) => 1.32 MiB
  content-script.js (1.35 MiB) => 1.33 MiB